lib/react_client.dart

```diff
@@ -476,72 +476,24 @@ final ReactDartInteropStatics _dartInteropStatics = (() {
       handleRender: allowInterop(handleRender));
 })();
 
-// TODO custom adapter for over_react to avoid typedPropsFactory usages?
-class JsComponent2Adapter extends Component2Adapter {
-  // TODO find a way to inject this better
-  final ReactComponent jsThis;
+abstract class _ReactDartInteropStatics2 {
+  // TODO 3.1.0-wip expose for testing?
+  static final zone = Zone.root;
 
-  JsComponent2Adapter({
-    this.jsThis,
-  });
-
-  @override
-  void forceUpdate(SetStateCallback callback) {
-    if (callback == null) {
-      jsThis.forceUpdate();
-    } else {
-      jsThis.forceUpdate(allowInterop(callback));
-    }
-  }
-
-  @override
-  void setState(Map newState, SetStateCallback callback) {
-    // Short-circuit to match the ReactJS 16 behavior of not re-rendering the component if newState is null.
-    if (newState == null) return;
-
-    dynamic firstArg = jsBackingMapOrJsCopy(newState);
-
-    if (callback == null) {
-      jsThis.setState(firstArg);
-    } else {
-      jsThis.setState(firstArg, allowInterop(([_]) {
-        callback();
-      }));
-    }
-  }
-
-  @override
-  void initializeState(Map state) {
-    dynamic jsState = jsBackingMapOrJsCopy(state);
-    jsThis.state = jsState;
+  static void _updatePropsAndStateWithJs(Component2 component, JsMap props, JsMap state) {
+    component
+      ..props = new JsBackedMap.backedBy(props)
+      ..state = new JsBackedMap.backedBy(state);
   }
 
-  @override
-  void setStateWithUpdater(StateUpdaterCallback stateUpdater, SetStateCallback callback) {
-    final firstArg = allowInterop((jsPrevState, jsProps, [_]) {
-      return jsBackingMapOrJsCopy(stateUpdater(
-        new JsBackedMap.backedBy(jsPrevState),
-        new JsBackedMap.backedBy(jsProps),
-      ));
-    });
-
-    if (callback == null) {
-      jsThis.setState(firstArg);
-    } else {
-      jsThis.setState(firstArg, allowInterop(([_]) {
-        callback();
-      }));
-    }
+  static void _updateContextWithJs(Component2 component, dynamic jsContext) {
+    component.context = _unjsifyNewContext(jsContext);
   }
-}
-
-final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
-  final zone = Zone.current;
 
   /// Wrapper for [Component.getInitialState].
-  Component2 initComponent(ReactComponent jsThis, ComponentStatics2 componentStatics) => zone.run(() {
+  static Component2 initComponent(ReactComponent jsThis, ComponentStatics2 componentStatics) => //
```

Comment:
Yean, it can be used to force dartfmt to insert a line break. I put these here because IMO it was easier to read to have all the `zone.run` calls on their own lines, as opposed to having just some of them.

> It seems to break pattern on line 550.
Good point; I'll add it to all of them for consistency's sake

I can also add a comment that makes it clearer what it's doing:
```dart
// dartfmt
```